devtools/x-lint/src/lib.rs

```diff
@@ -50,8 +50,9 @@ impl<'l, 'a> LintFormatter<'l, 'a> {
     }
 
     /// Writes a new lint message to this formatter.
-    pub fn write(&mut self, message: LintMessage) {
-        self.messages.push((self.source, message));
+    pub fn write(&mut self, level: LintLevel, message: impl Into<Cow<'static, str>>) {
```

Comment:
My favorite String type :D 

(I'm not joking I really do love `Cow<'stataic, str>`)

Reply:
I'll refactor it into `Cow<'l, str>` at some point :)